test/hotspot/jtreg/compiler/vectorapi/VectorMaskCompareNotTest.java

```diff
@@ -0,0 +1,1259 @@
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.NE, fa, fnan, (m) -> { return F_SPECIES.maskAll(true).xor(m); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.NE, fa, fnan);
+    }
+
+    @Test
+    @IR(counts = { IRNode.XOR_V_MASK, "= 0",
+                   IRNode.XOR_V, "= 0",
+                   IRNode.VECTOR_MASK_CMP, "= 2" },
+        applyIfCPUFeatureOr = { "asimd", "true", "avx", "true", "rvv", "true" })
+    public static void testCompareEQMaskNotFloatPositiveInfinity() {
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.EQ, fa, fpinf, (m) -> { return m.not(); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.EQ, fa, fpinf);
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.EQ, fa, fpinf, (m) -> { return F_SPECIES.maskAll(true).xor(m); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.EQ, fa, fpinf);
+    }
+
+    @Test
+    @IR(counts = { IRNode.XOR_V_MASK, "= 0",
+                   IRNode.XOR_V, "= 0",
+                   IRNode.VECTOR_MASK_CMP, "= 2" },
+        applyIfCPUFeatureOr = { "asimd", "true", "avx", "true", "rvv", "true" })
+    public static void testCompareNEMaskNotFloatPositiveInfinity() {
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.NE, fa, fpinf, (m) -> { return m.not(); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.NE, fa, fpinf);
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.NE, fa, fpinf, (m) -> { return F_SPECIES.maskAll(true).xor(m); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.NE, fa, fpinf);
+    }
+
+    @Test
+    @IR(counts = { IRNode.XOR_V_MASK, "= 0",
+                   IRNode.XOR_V, "= 0",
+                   IRNode.VECTOR_MASK_CMP, "= 2" },
+        applyIfCPUFeatureOr = { "asimd", "true", "avx", "true", "rvv", "true" })
+    public static void testCompareEQMaskNotFloatNegativeInfinity() {
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.EQ, fa, fninf, (m) -> { return m.not(); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.EQ, fa, fninf);
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.EQ, fa, fninf, (m) -> { return F_SPECIES.maskAll(true).xor(m); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.EQ, fa, fninf);
+    }
+
+    @Test
+    @IR(counts = { IRNode.XOR_V_MASK, "= 0",
+                   IRNode.XOR_V, "= 0",
+                   IRNode.VECTOR_MASK_CMP, "= 2" },
+        applyIfCPUFeatureOr = { "asimd", "true", "avx", "true", "rvv", "true" })
+    public static void testCompareNEMaskNotFloatNegativeInfinity() {
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.NE, fa, fninf, (m) -> { return m.not(); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.NE, fa, fninf);
+        testCompareMaskNotFloat(F_SPECIES, VectorOperators.NE, fa, fninf, (m) -> { return F_SPECIES.maskAll(true).xor(m); });
+        verifyResultsFloat(F_SPECIES, VectorOperators.NE, fa, fninf);
+    }
```

Comment:
Do you have test cases for the cases other than `EQ` and `NE`? After all, we don't that someone accidentally messes with the logic you implemented later and we don't notice the bug ;)